src/providers/dime/common/crypto.c

```diff
@@ -620,6 +620,7 @@ _generate_ed25519_keypair(void)
         PUSH_ERROR_OPENSSL();
         _secure_wipe(result, sizeof(ED25519_KEY));
         free(result);
+        result = NULL;
```

Comment:
Are you sure that this is necessary? The `RET_ERROR_PTR` macro below does not return `result`, therefore it looks safe to me. cppcheck can easily be wrong at this: as soon as it interprets `RET_ERROR_PTR` as an ordinary function call, it comes to the conclusion that `result` is still used, but that interpretation is already wrong.